### PR TITLE
[Fix-15771] Fix normal user can grant project permission

### LINKED ISSUE #15771: [Bug] [API] The user just can be view owner project in the default.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

The user can be view other user project used by current vulnerability.

### What you expected to happen

First：use the admin user create a project
Second：the normal user view projects, can't looked project
Third：get the permission and view the project

### How to reproduce

call it by interface

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by silentxingtian**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by qingwli** [comment]
Hi @silentxingtian , how about this method?

```
 /**
     * grant project
     *
     * @param loginUser  login user
     * @param userId     user id
     * @param projectIds project id array
     * @return grant result code
     */
    @Override
    @Transactional
    public Map<String, Object> grantProject(User loginUser, int userId, String projectIds) {
        Map<String, Object> result = new HashMap<>();
        result.put(Constants.STATUS, false);
        if (resourcePermissionCheckService.functionDisabled()) {
            putMsg(result, Status.FUNCTION_DISABLED);
            return result;
        }
        // check exist
        User tempUser = userMapper.selectById(userId);
        if (tempUser == null) {
            log.error("User does not exist, userId:{}.", userId);
            putMsg(result, Status.USER_NOT_EXIST, userId);
            return result;
        }
        
```
        
Looks like not checking is admin too. Could you fix this?

**Comment by silentxingtian** [comment]
> Hi @silentxingtian , how about this method?
> 
> ```
>  /**
>      * grant project
>      *
>      * @param loginUser  login user
>      * @param userId     user id
>      * @param projectIds project id array
>      * @return grant result code
>      */
>     @Override
>     @Transactional
>     public Map<String, Object> grantProject(User loginUser, int userId, String projectIds) {
>         Map<String, Object> result = new HashMap<>();
>         result.put(Constants.STATUS, false);
>         if (resourcePermissionCheckService.functionDisabled()) {
>             putMsg(result, Status.FUNCTION_DISABLED);
>             return result;
>         }
>         // check exist
>         User tempUser = userMapper.selectById(userId);
>         if (tempUser == null) {
>             log.error("User does not exist, userId:{}.", userId);
>             putMsg(result, Status.USER_NOT_EXIST, userId);
>             return result;
>         }
>         
> ```
> 
> Looks like not checking is admin too. Could you fix this?

ok, I will check all method and fix it

**Comment by silentxingtian** [comment]
i had fix grant project ,udf ,datasource method ,please check it again

**Review by SbloodyS — DISMISSED** [review]
Generally LGTM. We should hide these options in the front-end UI as improvement.

**Comment by SbloodyS** [comment]
@silentxingtian Please fix UT.

**Comment by silentxingtian** [comment]
> normal

this method need to add admin user check ,i had fix it.

**Comment by wangxj3** [comment]
LGTM

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java [hunk 1]
```diff
@@ -556,6 +556,12 @@ public Map<String, Object> grantProjectWithReadPerm(User loginUser, int userId,
             putMsg(result, Status.FUNCTION_DISABLED);
             return result;
         }
+
```
Comment (wangxj3) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:559]: LGTM

**Review by wangxj3 — APPROVED** [review]
LGTM

**Review by wangxj3 — APPROVED** [review]

**Review by davidzollo — APPROVED** [review]
LGTM
Thanks for your first contribution. I'm looking forward to your deep participation.

**Pull request merged**